figures/scripts/fig1-sample-distribution.R

```diff
@@ -0,0 +1,340 @@
+                    broad_histology,
+                    broad_histology_display),
+             by = "broad_histology") %>%
+  distinct() %>%
+  # Tack on the cancer group sample size
+  group_by(cancer_group) %>%
+  mutate(cancer_group_n = sum(n))
+
+# Set the y coordinate for the label based on the counts
+cancer_group_counts_df <- experimental_strategy_df %>%
+  select(broad_histology_display,
+         cancer_group,
+         cancer_group_n) %>%
+  distinct() %>%
+  group_by(broad_histology_display) %>%
+  mutate(y_coord = ceiling(cancer_group_n + (max(cancer_group_n) * 0.05)))
+
+
+# We need to handle cases where there is only one cancer group in the broad
+# histology display group otherwise the y coordinate for the label is way far
+# out
+broad_histology_counts <- cancer_group_counts_df %>%
+  count(broad_histology_display, name = "broad_histology_n")
+
+# And final data frame for plotting
+experimental_strategy_df <- cancer_group_counts_df %>%
+  # Add the broad histology counts required for the mutate step
+  left_join(broad_histology_counts) %>%
+  mutate(y_coord = case_when(
+    # When there's only one group, we avoid doubling the y-coordinate
+    broad_histology_n == 1 ~ (cancer_group_n * 1.05),
+    TRUE ~ y_coord
+  )) %>%
+  select(-broad_histology_n) %>%
+  left_join(experimental_strategy_df) %>%
+  select(-broad_histology)
+
+# We'll order the panels by the display order in the palettes data frame
+broad_histologies <- palette_df %>%
+  select(broad_histology_display, broad_histology_order) %>%
+  distinct() %>%
+  arrange(broad_histology_order) %>%
+  pull(broad_histology_display)
+
+#### Tumor descriptor stacked bar plots ----------------------------------------
+
+create_descriptor_plot <- function(broad_histology_label) {
+
+  # Create a stacked bar plot that shows the proportion of tumors in a cancer
+  # group in each tumor descriptor category
+  descriptor_plot <- tumor_descriptor_df %>%
+    filter(broad_histology_display == broad_histology_label) %>%
+    ggplot(aes(x = cancer_group_label,  # includes n
+               y = proportion,
+               fill = tumor_descriptor)) +
+    geom_bar(color = "#000000", position = "fill", stat = "identity") +
+    scale_fill_manual(values = tumor_descriptor_palette) +
+    theme_pubr() +
+    theme(axis.text.x = element_text(angle = 90,
+                                     vjust = 0.5,
+                                     hjust = 1),
+          plot.title = element_text(hjust = 0.5,
+                                    face = "bold",
+                                    size = 9)) +
+    labs(x = "",
+         y = "Proportion",
+         title = broad_histology_label) +
+    guides(fill = FALSE)
+
+}
+
+# Apply to each broad histology display group
+descriptor_plot_list <- purrr::map(broad_histologies,
+                                   ~ create_descriptor_plot(.x))
+
+# Using wrap_plots() with all 12 yields some bad results, so here we break
```

Comment:
Looking at these plots with all the repeated y axes... 

Can we use `facet_wrap(~broad_histologies, scales = "free_x")` to avoid the repeated labels? I assume we would still need to break into groups of 4 to deal with whatever problem the 12 plots created.

Alternatively, we could get even fancier and do something like  `purrr::map2(broad_histologies, c(TRUE, FALSE, FALSE, FALSE), ~create_descriptor_plot(.x, .y)` where the second argument determines whether the y axis is labeled in the individual plots.